core/state/snapshot/snapshot.go

```diff
@@ -560,14 +560,20 @@ func diffToDisk(bottom *diffLayer) *diskLayer {
 
 	// Destroy all the destructed accounts from the database
 	for hash := range bottom.destructSet {
+		oldAccount, err := base.account(hash, true)
 		// Skip any account not covered yet by the snapshot
-		if base.genMarker != nil && bytes.Compare(hash[:], base.genMarker) > 0 {
+		if errors.Is(err, ErrNotCoveredYet) {
 			continue
 		}
-		// Remove all storage slots
 		rawdb.DeleteAccountSnapshot(batch, hash)
 		base.cache.Set(hash[:], nil)
 
+		if err == nil && (oldAccount == nil || oldAccount.Root == nil) {
```

Comment:
For the sake of a minimal understandable diff, I'd put here the call to base.account.
Also, either handle err != nil in some way or comment why you don't need to.

Reply:
I've added a warning if err != nil and a comment explaining that it falls through to iterating over and removing storage in case there is any